server/tools/hatohol-house-keeper

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+import sys
+import os
+import argparse
+import MySQLdb
+import time
+
+
+def finish_if_require_option_is_fault(args):
+    if args.date and args.number:
+        print 'You can not set -d (--date) and -n (--number) option together.'
+        sys.exit(os.EX_USAGE)
+    if not args.date and not args.number and not args.stat:
+        print 'You need to set -d (--date) or -n (--number) or --stat option.'
+        sys.exit(os.EX_USAGE)
+    if (args.date or args.number) and not args.table:
+        print 'You need to set table name using --table option.'
+
+
+def show_number_of(table_name, cursor):
+    query = 'SELECT count(*) FROM %s' % table_name
+    cursor.execute(query)
+    row = cursor.fetchone()
+
+    print '%s: %d' % (table_name, row[0])
+
+
+def show_status(cursor):
+    show_number_of('events', cursor)
+    show_number_of('action_logs', cursor)
+    sys.exit(os.EX_OK)
+
+
+def return_unix_time(date):
+    format = '%Y-%m-%d %H:%M:%S'
+    date += ' 00:00:00'
+    return int(time.mktime(time.strptime(date, format)))
+
+
+def run_sql_query(db, cursor, query):
+    try:
+        cursor.execute(query)
+        db.commit()
+    except:
+        db.rollback()
+
+
+def delete_unnecessary_records_by_date(table_name, date, db, cursor):
+    if 'events' in table_name:
+        time_sec = return_unix_time(date)
+        query = 'DELETE FROM events WHERE time_sec < %d' % time_sec
+    elif 'action_logs' in table_name:
+        query = 'DELETE FROM action_logs WHERE queuing_time < %s' % date
```

Comment:
Do all the action log records have valid queuing_time ?
This is just a question. I forget the behavior.